tests/components/dio_chacon/test_config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Test the dio_chacon config flow."""
+
+import logging
+from unittest.mock import AsyncMock, patch
+
+from dio_chacon_wifi_api.exceptions import DIOChaconAPIError, DIOChaconInvalidAuthError
+
+from homeassistant import config_entries
+from homeassistant.components.dio_chacon.const import DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_UNIQUE_ID, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def test_show_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == config_entries.SOURCE_USER
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the form."""
+
+    with patch(
+        "dio_chacon_wifi_api.DIOChaconAPIClient.get_user_id",
+        return_value="dummy-user-id",
+    ) as mock_dio_chacon_client:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={
+                CONF_USERNAME: "dummylogin",
+                CONF_PASSWORD: "dummypass",
+                CONF_UNIQUE_ID: "dummy-user-id",
+            },
+        )
+
+        _LOGGER.debug("Test result after init : %s", result)
+
+    assert len(mock_dio_chacon_client.mock_calls) == 1
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Dio Chacon dummylogin"
+    assert result["data"] == {
+        CONF_USERNAME: "dummylogin",
+        CONF_PASSWORD: "dummypass",
+        CONF_UNIQUE_ID: "dummy-user-id",
```

Comment:
This isn't here in real life, so lets's remove the unique id from the result data.

Lets add a check if the `result["result"].unique_id` equals to that unique id

Reply:
Done